### DISCUSSION TITLE
Covert Spacy annotation to Flair BIO format (Overlap Issue)

### DISCUSSION THREAD

**Discussion by wahid18benz** [explosion/spaCy#6513] [Help: Coding & Implementations] [answered]
Hello, I'm trying to convert Spacy annotation to Flair BIO Format, I'm facing an overlapping problem between entities!
Here is and exemple of the code:

`

        import spacy

        from spacy.gold import biluo_tags_from_offsets

        nlp = spacy.load('en_core_web_sm')

        TRAIN_DATA =  [ 

        ("PRICE INCREASED EVERY 6 HOURS", {"entities": [(16, 29, "TIME"),(22, 29, "TIME")]}),
        ("DISCOUNT SALES ANNOUNCED FOR  2 TIMES DAILY", {"entities": [(30, 43, "ORG"),(38, 43, "ORG")]})

        ]

        docs = []

        for text, annot in TRAIN_DATA:

        doc = nlp(text)

        biluo = biluo_tags_from_offsets(doc, annot['entities'])

        for word,tag in zip(doc, biluo):

           print(word, tag)
`
**"Error: Trying to set conflicting doc.ents: '(16, 29, 'TIME')' and '(22, 29, 'TIME')'. A token can only be part of one entity, so make sure the entities you're setting don't overlap."**
is there any solution for the overlapping problem ?

**Comment by adrianeboyd** [comment, marked as answer]
No, the basic constraint is that the IOB format can't represent overlapping entities.

A spacy `Doc` is using IOB internally to store entities, so it can't store overlapping entities like these. And even if it could, you couldn't export overlapping entity annotation as IOB tags.

You will need to filter or modify the spans in your annotations so that none of them overlap before trying to convert them to IOB.
